examples/bubbling.rs

```diff
@@ -0,0 +1,127 @@
+//! [`ForwardedEvent`] trait you implemented on your custom event to convert the `PointerEvent` into
+//! your custom event.
+//!
+//! In other words, the `forward_events` function is really a helper function that just inserts a
+//! predefined `EventListener` component on your entity.
+
+use bevy::prelude::*;
+use bevy_mod_picking::prelude::{
+    backends::raycast::{PickRaycastSource, PickRaycastTarget, RaycastPlugin},
+    *,
+};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_plugins(DefaultPickingPlugins)
+        .add_plugin(RaycastPlugin)
+        .add_plugin(DebugEventsPlugin)
+        .add_startup_system(setup)
+        .add_system(DeleteMe::handle_events)
+        .add_system(GreetMe::handle_events)
+        .run();
+}
+
+// We want to implement a feature that will allow us to click on a cube to delete it. To do this,
+// we'll start by making an event we can send when we want to delete an entity.
+struct DeleteMe(Entity);
+// We're going to use the event forwarding feature of this crate to send a `DeleteMe` event when the
+// entity is clicked. To be able to forward events, we need to implement the `ForwardedEvent` trait
+// on our custom `DeleteMe` event.
+//
+// All we're doing is defining how to take a pointer event and turn it into our custom event.
+impl ForwardedEvent for DeleteMe {
+    fn from_data<E: IsPointerEvent>(event_data: &PointerEventData<E>) -> DeleteMe {
+        // Note that we are using the `target()` entity here, not the listener entity! The target is
+        // the child that the event was originally called on, whereas the listener is the ancestor
+        // that was listening for the event that bubbled up from the target.
+        DeleteMe(event_data.target())
+        // Why is this useful? It allows us to add an event listener once on the parent entity, yet
+        // it can trigger actions specific to the child that was interacted with! Instead of needing
+        // to add an event listener on every child, we can just stick one on the parent, and any
+        // events that happen on the children will bubble up the the parent and be handled there.
+    }
+}
+impl DeleteMe {
+    // Here we will implement the system that does something with our `DeleteMe` events.
+    fn handle_events(mut commands: Commands, mut delete: EventReader<DeleteMe>) {
+        for event in delete.iter() {
+            commands.entity(event.0).despawn_recursive();
+            info!("I deleted the thing!");
+        }
+    }
+}
+
+// Same concept as the `DeleteMe` event, but just says "Hello!" to the entity.
+struct GreetMe(Entity);
+impl ForwardedEvent for GreetMe {
+    fn from_data<E: IsPointerEvent>(event_data: &PointerEventData<E>) -> GreetMe {
+        GreetMe(event_data.target())
+    }
+}
+impl GreetMe {
+    fn handle_events(mut greet: EventReader<GreetMe>) {
+        for event in greet.iter() {
+            info!("Hello {:?}!", event.0);
```

Comment:
Might be nice to print both the target and listener entities so it's more obvious where the events are coming from.